source/common/http/http1/conn_pool.cc

```diff
@@ -313,6 +322,11 @@ ConnPoolImpl::ActiveClient::~ActiveClient() {
   parent_.host_->stats().cx_active_.dec();
   conn_length_->complete();
   parent_.host_->cluster().resourceManager(parent_.priority_).connections().dec();
+
+  if (!connection_succeeded_) {
+    // we failed to decrement pendingConnections previously, so do it now
```

Comment:
hyper hyper nit: all other comments in this file (and elsewhere) are properly constructed sentences (initial upper case + punctuation)